src/system/system.services.ts

```diff
@@ -52,3 +75,73 @@ function getBuildInfo(bindings: AppBindings): BuildInfo {
     return buildInfo;
   }
 }
+
+export function setMaintenanceWindow(
+  ctx: AppBindings,
+  request: AdminSetMaintenanceWindowRequest,
+): E.Effect<
+  boolean,
+  | DocumentNotFoundError
+  | DataValidationError
+  | PrimaryCollectionNotFoundError
+  | DatabaseError
+> {
+  return pipe(
+    E.succeed(request),
+    E.flatMap((request) => {
+      if (request.id !== ctx.node.identity.did) {
+        return E.fail(
+          new DataValidationError({
+            issues: ["DID prohibited"],
+            cause: request,
+          }),
+        );
+      }
+
+      const now = new Date();
```

Comment:
I've added the temporal-polyfill ... take a look it makes working with dates much cleaner.